source/edk.c

```diff
@@ -101,3 +101,9 @@ bool aws_cryptosdk_edk_list_is_valid(const struct aws_array_list *edk_list) {
     return true;
 #endif /* AWS_DEEP_CHECKS == 1 */
 }
+
+bool aws_cryptosdk_empty_edk_list_is_valid(const struct aws_array_list *edk_list) {
+    AWS_PRECONDITION(edk_list->length == 0);  // ensure that the list is empty
+    return AWS_OBJECT_PTR_IS_READABLE(edk_list) && aws_array_list_is_valid(edk_list) &&
+           (edk_list->item_size == sizeof(struct aws_cryptosdk_edk));
+}
```

Comment:
In this case, `AWS_OBJECT_PTR_IS_READABLE(edk_list)` will always be true. In fact, you need to add it as a precondition, so you'll avoid the possibility of a null-pointer dereference in `edk_list->length`. Do we consider `edk_list == NULL` as a valid empty edk_list? If not, I'd suggest:
```c
bool aws_cryptosdk_empty_edk_list_is_valid(const struct aws_array_list *edk_list) {
    AWS_PRECONDITION(AWS_OBJECT_PTR_IS_READABLE(edk_list));
    AWS_PRECONDITION(edk_list->length == 0);
    return aws_array_list_is_valid(edk_list) &&
           (edk_list->item_size == sizeof(struct aws_cryptosdk_edk));
}
```
We can also remove the comment, since the code is pretty clear.

Reply:
You're right. We do not consider edk_list == NULL as a valid empty list, so I've made the changes you suggested.